libraries/AP_Math/vector3.h

```diff
@@ -192,9 +192,15 @@ class Vector3
     // rotate vector by angle in radians in xy plane leaving z untouched
     void rotate_xy(float rotation_rad);
 
-    // return xy components of a vector3
-    Vector2<T> xy();
-    
+    // return xy components of a vector3 as a vector2.
+    // this returns a reference to the original vector3 xy data
+    const Vector2<T> &xy() const {
+        return *(const Vector2<T> *)this;
+    }
+    Vector2<T> &xy() {
```

Comment:
why shouldn't this one be const too ? it doesn't modify the base object

Reply:
its returning a non-const version of the object, so can't be const